### PR TITLE
fixes #85 - use a dedicated directory for harvester logs in geonetwork

### LINKED ISSUE #85: geonetwork harvester logs are stored in geonetwork_datadir
See https://github.com/georchestra/georchestra/issues/1850

### PR DESCRIPTION
**Pull request by severo**
Code by https://github.com/landryb, see:

  https://github.com/georchestra/georchestra/issues/1850#issuecomment-416949322

### CODE REVIEW

**Comment by severo** [comment]
I think this commit must not be merged into docker-master as is.

File: geonetwork/log4j/log4j.xml [hunk 1]
```diff
@@ -11,6 +11,13 @@
             <param name="ConversionPattern" value="%d{ISO8601} %-5p [%c] - %m%n" />
         </layout>
     </appender>
+    <appender name="harvesterAppender" class="org.apache.log4j.FileAppender">
+        <param name="Threshold" value="ALL" />
+        <param name="File" value="/tmp/harvester/harvester.log" />
```
Comment (fvanderbiest) [inline geonetwork/log4j/log4j.xml]: If  `/tmp/harvester/` dir does not exist at runtime, there's a chance this fails.
Please use `/tmp/geonetwork-harvester.log` instead

**Comment by fvanderbiest** [comment]
> I think this commit must not be merged into docker-master as is.

Indeed.

File: geonetwork/log4j/log4j.xml (hunk 1 above)
Comment (severo) [reply geonetwork/log4j/log4j.xml]: You're right, much better -> https://github.com/georchestra/datadir/pull/86/commits/5c8ca82030112072353608cd3f94846ec58009ef

**Comment by fvanderbiest** [comment]
Can you propose something for the merge of `master` into `docker-master` ?
Thanks.

**Comment by severo** [comment]
@fvanderbiest do you think it's necessary to change anything in the docker-master branch (currently, the logs produced by `geonetwork.harvester` are sent to the console, the same way as all the logs)?
I don't know how these logs are generally consumed and treated later, but I imagine two ways:
 - don't change anything
 - add an information inside the log text (something like `- harvester`) and output to the console.

What do you recommend?

**Comment by fvanderbiest** [comment]
I think we can keep the log4j configuration from docker-master as it currently is.

Please check that this does not break harvest logs download from the /geonetwork/srv/fre/admin.console#/harvest page when using a dockerized geonetwork instance

**Pull request merged**